lib/index.ts

```diff
@@ -0,0 +1,27 @@
+import { clientInfo } from './environment'
+
+declare var PercyAgent: any;
+declare var Nightmare: any;
+
+/**
+ * A function to take a Percy snapshot from a NightmareJS test. To use in your tests:
+ *   const { percySnapshot } = require('@percy/nightmare')
+ *   [...]
+ *   nightmare
+ *     .goto(<yourtesturl>)
+ *     .use(percySnapshot(<your snapshot name>), <maybe options>))
+ *
+ * @param name Name of the snapshot that we're taking.
+ * @param options Additional options, e.g. '{widths: [300, 600, 1000]}'. Optional.
+ */
+export function percySnapshot(name: string, options: any = {}) {
+  return function (nightmare: any) {
+    nightmare
+      .inject('js', 'node_modules/@percy/nightmare/node_modules/@percy/agent/dist/public/percy-agent.js')
+      .evaluate(function (name: string, options: any, clientInfo: string) {
+        name = name || document.title
```

Comment:
@djones a thing to note: if no defined snapshot name is passed, i use the document title. This means that if the caller doesn't pass any names, we're likely to get identical snapshot names for all snapshots (because in most apps the document title tends to be something static and generic about the app). Unfortunately at this point in the code we don't have access to anything about the tests, so it is hard to make up anything more specific to use as the snapshot name.

Reply:
Updated to make `name` a required parameter, and throw an `Error` if no name is passed. In the error message I suggest using `this.test.fullTitle()` if the user is in Mocha.